tests/integration/__tests__/cache.js

```diff
@@ -44,6 +44,15 @@ describe("--cache option", () => {
     );
   });
 
+  it("throws error when use with --stdin-filepath", async () => {
+    const { stderr } = await runPrettier(
+      dir,
+      ["--cache", "--stdin-filepath", "foo.js"],
+      { input: "const a = a;" }
+    );
+    expect(typeof stderr).toBe("string");
```

Comment:
Isn't `stderr` always string? We should either ignore the `--cache` flag or print something like ` can't use `--cache` when formatting stdin`.